ucsm_apis/admin/ldap.py

```diff
@@ -0,0 +1,875 @@
+    handle.remove_mo(mo)
+    handle.commit()
+
+
+def ldap_provider_group_provider_add(handle, group_name, name,
+                                     order="lowest-available",
+                                     descr=None, **kwargs):
+    """
+    adds provider to ldap provider group
+
+    Args:
+        handle (UcsHandle)
+        group_name (string): ldap provider group name
+        name (string): name
+        order (string): order
+        descr (string): descr
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        AaaProviderRef : Managed Object
+
+    Raises:
+        UcsOperationError: If AaaProviderGroup or AaaProvider is not present
+
+    Example:
+        ldap_provider_group_provider_add(handle,
+                                        group_name="test_ldap_provider_group",
+                                        name="test_provider",
+                                        order="1")
+    """
+
+    from ucsmsdk.mometa.aaa.AaaProviderRef import AaaProviderRef
+
+    group_dn = _base_dn + "/ldap-ext/providergroup-" + group_name
+    group_mo = handle.query_dn(group_dn)
+    if not group_mo:
+        raise UcsOperationError("ldap_provider_group_provider_add",
+                                 "Ldap Provider Group does not exist.")
+
+    provider_dn = _base_dn + "/ldap-ext/provider-" + name
+    provider_mo = handle.query_dn(provider_dn)
+    if not provider_mo:
+        raise UcsOperationError("ldap_provider_group_provider_add",
+                                 "Ldap Provider does not exist.")
+
+    mo = AaaProviderRef(parent_mo_or_dn=group_mo,
+                        name=name,
+                        order=order,
+                        descr=descr)
+
+    mo.set_prop_multiple(**kwargs)
+
+    handle.add_mo(mo, True)
+    handle.commit()
+    return mo
+
+
+def ldap_provider_group_provider_get(handle, group_name, name,
+                                    caller="ldap_provider_group_provider_get"):
+    """
+    Gets provider for ldap provider group
+
+    Args:
+        handle (UcsHandle)
+        group_name (string): ldap provider group name
+        name (string): name
+
+    Returns:
+        AaaProviderRef : Managed Object OR None
+
+    Example:
+        ldap_provider_group_provider_get(handle,
+                                         group_name="test_ldap_provider_group",
+                                         name="test_provider",
+                                         order="1")
+    """
+    provider_group_dn = _ldap_dn+ "providergroup-" + group_name
+    provider_dn = provider_group_dn + "/provider-ref-" + name
+    mo = handle.query_dn(provider_dn)
+    if mo is None:
+        raise UcsOperationError(caller,
+                                "Ldap Provider '%s' does not exist" % dn)
```

Comment:
`dn` undefined. Should it be mo.dn ?
Should we print the name instead of the dn ?